pkg/operator/operator_config.go

```diff
@@ -646,11 +646,59 @@ func (r *operatorConfigReconciler) makeAlertmanagerConfigs(ctx context.Context,
 		if err != nil {
 			return nil, nil, fmt.Errorf("cannot build regex from service name %q: %w", am.Name, err)
 		}
-		cfg.RelabelConfigs = append(cfg.RelabelConfigs, &relabel.Config{
-			Action:       relabel.Keep,
-			SourceLabels: prommodel.LabelNames{"__meta_kubernetes_endpoints_name"},
-			Regex:        svcNameRE,
-		})
+		cfg.RelabelConfigs = append(cfg.RelabelConfigs,
```

Comment:
Should we ensure there is a dynamic (e.g. config based) switch off for this? e.g. when we block some customer with this?

Reply:
The design rejected putting this change behind a feature flag, and also suggests that the creation of a feature flag would increase confusion and the maintenance burden for TargetLabels.

An MSA is planned to notify users of this upcoming change and allow them to adjust accordingly.

Reply:
I believe the idea is that users before this change are unimpacted since there may be customers already using a label named `workload_controller`. The `PodMonitoring.targetLabels` field is essentially the switch. This PR is missing that.

@bernot-dev let's update the Defaulter, such that if the customer created a new PodMonitoring (e.g. did not set a `targetLabels` value), we add `workload_controller` in the list. Then, only add these additional configurations if it is specified (see `relabelingsForMetadata`).